### DISCUSSION TITLE
Scheduling an Amber Dispatch, and shutting down a Dispatch - using Home Assistant - or any other means

### DISCUSSION THREAD

**Discussion by geoffw333** [amberelectric/public-api#207] [Home Assistant]
Hey, newb here, please go easy on me.

For starts - I have a Sungrow setup - 38kwh of storage, Hybrid 3 phase inverter, and 8.4kw of generation via Jinko panels. I use Home Assistant for lots of stuff, the Sungrow instrumentation is integrated into HA, Philips Hue Lights, gates, security cameras, property irrigation system etc etc etc 

I just jumped onto Amber as a provider, and am learning the ropes. I have the ha "show data" API and instrumentation working, but what I am missing is the methods to go and call the Dispatch functions. There seem to be no actual methods available to do that. 

Maybe I am missing something ? Should I be getting it going in an easier to handle environment (jscript, or even excel ? ) I see you can get a much richer data set using the json/API data, than just landing it in ha. 

Any tips/suggestions welcome.

Thanks

Geoff

**Comment by rcruikshank** [comment]
I would investigate [EMHASS](https://community.home-assistant.io/t/emhass-an-energy-management-for-home-assistant/338126) as an energy management addon to Home Assistant that incorporates not only battery and PV management but also deferable load management in sync with PV and battery that HA is able to control. It uses linier or MPC control to calculate a daily or constant plan (up to you) that will get the most out of the available PV, battery and tariff fluctuations. It also has machian learning algorithms to analyse consumption (or nay) data for better processing.

Also, I would consider using Node-Red to build the interface between EMHASS and HA.  It makes it easy to develop complex control system. My config is [here](https://cruikshank-my.sharepoint.com/:f:/g/personal/robert_cruikshank_onmicrosoft_com/EliM7mPksJpBtEaYwf7q920BxHMk4ur2aXqf_6aQB5AO1A?e=5zmaVB).

As for controlling Sungrow systems, this is the advice from ChatGPT. I'd ask Sungrow for the documentation re API control for the equipment you have. I use the available REST API for my Sonnen battery which allows precise control over charging and discharging:
_

> Sungrow, a prominent manufacturer of inverters and energy storage systems, does provide APIs and communication interfaces for their products, which include batteries. These APIs allow integration with energy management systems and enable control over various functionalities, such as charging and discharging.
> 
> However, the specific details and availability of these APIs can vary depending on the model and the specific requirements of your installation. Typically, these APIs might allow you to:
> 
> Monitor Battery Status: Check the current charge level, temperature, and other vital statistics.
> Control Charging and Discharging: Schedule or manually initiate charging and discharging cycles.
> Set Operational Parameters: Define parameters like charge and discharge rates, power limits, and operating modes.
> To access these APIs, you generally need to:
> 
> Refer to the Technical Documentation: The API details are usually included in the technical manuals or user guides provided by Sungrow.
> Contact Sungrow Support: For specific models or advanced functionalities, reaching out to Sungrow's technical support can provide you with the necessary documentation and support.
> Use Compatible Communication Protocols: Sungrow devices often use standard communication protocols such as Modbus TCP/IP, which can be integrated with many energy management systems.
> If you have a specific model in mind or particular requirements for your use case, it would be beneficial to check the exact product documentation or contact Sungrow directly for the most accurate and detailed information._

**Reply by geoffw333** [reply]
thanks for the response. I'll look at emhass - had a look at NodeRed - I like the abstraction, but don't want to load up more software at this stage. 

I have the Sungrow modbus mapping and reporting into ha working fine; the modbus mappings are a bit of a movieable feast. 

I have a call into amber, as that's the piece I want to address as priority

Cheers.
